### PR TITLE
[Xamarin.Android.Build.Tasks] Remove EmbeddedNativeLibrary from Mono.Data.Sqlite.

### LINKED ISSUE #2407: Remove the use of EmbeddedNativeLibrary from Mono.Posix and Mono.Data.Sqlite
We currently embed the required native libraries for Mono.Data.Sqlite and Mono.Posix into the assembly as an EmbeddedResource (`__AndroidNativeLibraries__.zip`). This is then extracted as part of the build process into the apk. This is the only actual change we have from the vanilla versions of both of these assemblies. 

We should update our build system to special case these two assemblies and just copy over/include the required native libraries from the installed Xamarin.Android system. The required files are

`libsqlite3_xamarin.so`
`libMonoPosixHelper.so`

both of which should end up in the `lib` directory of the Xamarin.Android MSBuild install directory. Currently only `libMonoPosixHelper.so` does.

### PR DESCRIPTION
**Pull request by dellis1972**
Fixes #2407

We currently embed the required native libraries for
Mono.Data.Sqlite and Mono.Posix into the assembly as
 an `EmbeddedResource` (__AndroidNativeLibraries__.zip).
This is then extracted as part of the build process
into the apk. This is the only actual change we have
from the vanilla versions of both of these assemblies.

This commit removes the `EmbeddedResource` items from
the assmeblies. It also updates sqlite-xamarin to place
its outputs into the

	`lib\xamarin-android\xbuild\Xamarin\Android\lib\{arch}\`

directories in the main output directory. This will make sure
we install them as part of the various installers we have.

We also update the MSBuild targets to check to see if these
assemblies are referenced and include the required `.so` files.

### CODE REVIEW

File: src/Mono.Posix/Mono.Posix.csproj [hunk 1]
```diff
@@ -258,18 +258,4 @@
...
-    <EmbeddedNativeLibrary Include="$(XANativeLibsDir)\armeabi-v7a\libMonoPosixHelper.so" Condition="$(AndroidSupportedTargetJitAbisForConditionalChecks.Contains (':armeabi-v7a:'))">
-      <Link>MonoPosixHelper\armeabi-v7a\libMonoPosixHelper.so</Link>
-    </EmbeddedNativeLibrary>
-    <EmbeddedNativeLibrary Include="$(XANativeLibsDir)\x86\libMonoPosixHelper.so" Condition="$(AndroidSupportedTargetJitAbisForConditionalChecks.Contains (':x86:'))">
-      <Link>MonoPosixHelper\x86\libMonoPosixHelper.so</Link>
-    </EmbeddedNativeLibrary>
-    <EmbeddedNativeLibrary Include="$(XANativeLibsDir)\x86_64\libMonoPosixHelper.so" Condition="$(AndroidSupportedTargetJitAbisForConditionalChecks.Contains (':x86_64:'))">
-      <Link>MonoPosixHelper\x86_64\libMonoPosixHelper.so</Link>
-    </EmbeddedNativeLibrary>
-  </ItemGroup>
```
Comment (jonathanpeppers) [inline src/Mono.Posix/Mono.Posix.csproj]: What actually puts `libMonoPosixHelper.so` into `bin\$(Configuration)\lib\xamarin.android\xbuild\Xamarin\Android\lib`?

**Comment by jonpryor** [comment]
As part of this PR, I think it would be worth exploring *completely removing* the `src/Mono.Data.Sqlite` directory, and instead updating `src/mono-runtimes` to add `Mono.Data.Sqlite.dll` to the mono bundle, a'la:

```diff
diff --git a/src/mono-runtimes/ProfileAssemblies.projitems b/src/mono-runtimes/ProfileAssemblies.projitems
index 7a95aa88..f1dbbb6d 100644
--- a/src/mono-runtimes/ProfileAssemblies.projitems
+++ b/src/mono-runtimes/ProfileAssemblies.projitems
@@ -160,6 +160,7 @@
     <MonoProfileAssembly Include="Mono.Btls.Interface.dll" />
     <MonoProfileAssembly Include="Mono.CompilerServices.SymbolWriter.dll" />
     <MonoProfileAssembly Include="Mono.CSharp.dll" />
+    <MonoProfileAssembly Include="Mono.Data.Sqlite.dll" />
     <MonoProfileAssembly Include="Mono.Data.Tds.dll" />
     <MonoProfileAssembly Include="Mono.Security.dll" />
     <MonoProfileAssembly Include="mscorlib.dll" />
```

I don't see anything within `Mono.Data.Sqlite.csproj` which would continue that we build it from source, other than the `@(Compile)` entry for `src\Mono.Android\Android.Runtime\CPUArchitecture.cs`, which doesn't make any sense to me (I can't see any use of the `CPUArchitecture` enum within `Mono.Data.Sqlite.dll`, and thus wonder why `CPUArchitecture.cs` is used *at all*.)

**Comment by dellis1972** [comment]
@jonpryor one thing we do have in Mono.Data.Sqlite is the `dll.config` which contains

```
<configuration>
  <dllmap dll="sqlite3" target="libsqlite3_xamarin.so" />
</configuration>
```

So we need to make sure that gets included next to the dll copied over from mono

**Comment by luhenry** [comment]
Why wasn't `src/Mono.Posix/Mono.Posix.csproj`  removed as well?

**Pull request merged**